### PR TITLE
Add functionality needed for notebook processing and XFDF generation

### PR DESCRIPTION
**Pull request by th-we**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -128,8 +139,93 @@ pub struct Line {
...
+        }
+    }
 
-#[derive(Default, Debug)]
+    fn length(&self) -> f32 {
+        let mut length = 0.0;
+        for (previous_index, point) in self.points[1..].iter().enumerate() {
+            length += point.distance(&self.points[previous_index]);
+        }
+        length
```
Comment (davidrusu) [inline src/lib.rs]:
```suggestion
        self.points.iter()
            .zip(self.points[1..].iter())
            .map(|(previous_point, point)| previous_point.distance(point))
            .sum()
```

**Review by davidrusu — COMMENTED** [review]
looks pretty nice so far 👍, did you want to add the actual XFDF generation in this PR or will that be in the future?

File: src/lib.rs (hunk 1 above)
Comment (th-we) [reply src/lib.rs]: Oh, really nice, this `zip()` method. Thanks for guiding me!

**Comment by th-we** [comment]
Many thanks for going over the code and making suggestions!  I really appreciate a more experienced rustacean looking over my still inexperienced shoulder.

XFDF generation is still not useful at all and needs quite a bit more work.  Do you suggest that I commit it anyway in this unusable state?

**Comment by th-we** [comment]
Any idea what causes the GitHub check to fail?  It just says "Error" without any further info. Could it be the many still unused functions?

**Comment by davidrusu** [comment]
The CI failures are unrelated, something with the nix build is not happy.

I'm going to merge this as I was going to do some work in the library soon

**Pull request merged**